<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>760</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>操作系统课程设计</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>算法演示</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QSplitter" name="splitter_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QSplitter" name="splitter">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>FIFO演示</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QTableWidget" name="FIFOTable">
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <property name="gridStyle">
                <enum>Qt::SolidLine</enum>
               </property>
               <property name="sortingEnabled">
                <bool>false</bool>
               </property>
               <property name="rowCount">
                <number>0</number>
               </property>
               <property name="columnCount">
                <number>0</number>
               </property>
               <attribute name="horizontalHeaderVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>LRU演示</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0">
              <widget class="QTableWidget" name="LRUTable">
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <attribute name="horizontalHeaderVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>LFU演示</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QTableWidget" name="LFUTable">
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <attribute name="horizontalHeaderVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>OPT演示</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="0">
              <widget class="QTableWidget" name="OPTTable">
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <attribute name="horizontalHeaderVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
          <widget class="QSplitter" name="splitter_2">
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <widget class="QGroupBox" name="groupBox_5">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="title">
             <string>参数设置</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="3" column="1">
              <widget class="QSpinBox" name="memTimeBox">
               <property name="suffix">
                <string> ns</string>
               </property>
               <property name="maximum">
                <number>99999</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>页号范围</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>缺页中断时间</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>快表访问时间</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="inputLenEdit">
               <property name="text">
                <string>20</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QCheckBox" name="checkBox">
               <property name="text">
                <string>启用快表</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>内存访问时间</string>
               </property>
              </widget>
             </item>
             <item row="7" column="0" colspan="2">
              <widget class="QTextEdit" name="textEdit">
               <property name="font">
                <font>
                 <family>Consolas</family>
                </font>
               </property>
               <property name="autoFormatting">
                <set>QTextEdit::AutoNone</set>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="pageNumBox">
               <property name="suffix">
                <string> Bit</string>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>20</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="misTimeBox">
               <property name="suffix">
                <string> ns</string>
               </property>
               <property name="maximum">
                <number>99999</number>
               </property>
               <property name="value">
                <number>105</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>访问页面数</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="frameLenEdit">
               <property name="text">
                <string>3</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QSpinBox" name="TLBTimeBox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="suffix">
                <string> ns</string>
               </property>
               <property name="maximum">
                <number>99999</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>驻留内存页面个数</string>
               </property>
              </widget>
             </item>
             <item row="8" column="0">
              <widget class="QPushButton" name="pushButton">
               <property name="text">
                <string>生成序列</string>
               </property>
              </widget>
             </item>
             <item row="8" column="1">
              <widget class="QPushButton" name="storeButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>保存</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_6">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="title">
             <string>启动</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_7">
             <item row="0" column="0">
              <widget class="QPushButton" name="initButton">
               <property name="text">
                <string>初始化</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="allStartButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>全部开始</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="FIFOButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>FIFO开始</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="LRUButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>LRU开始</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="LFUButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>LFU开始</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QPushButton" name="OPTButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>OPT开始</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_7">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="title">
             <string> 访问时间</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLabel" name="FIFOLabel">
               <property name="text">
                <string>FIFO:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="LRULabel">
               <property name="text">
                <string>LRU:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="LFULabel">
               <property name="text">
                <string>LFU:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="OPTLabel">
               <property name="text">
                <string>OPT:</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QGroupBox" name="groupBox_8">
            <property name="maximumSize">
             <size>
              <width>200</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="title">
             <string> 缺页率</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_8">
             <item row="0" column="0">
              <widget class="QLabel" name="FIFOLabel_2">
               <property name="text">
                <string>FIFO:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QProgressBar" name="FIFOProgressBar">
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="LRULabel_2">
               <property name="text">
                <string>LRU:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QProgressBar" name="LRUProgressBar">
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="LFULabel_2">
               <property name="text">
                <string>LFU:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QProgressBar" name="LFUProgressBar">
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="OPTLabel_2">
               <property name="text">
                <string>OPT:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QProgressBar" name="OPTProgressBar">
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string> 历史记录</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QTableWidget" name="tableWidget">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>缺页数演示</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <widget class="QPushButton" name="refreshButton">
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>刷新</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_12">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>当前输入序列:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="4">
         <widget class="QChartView" name="chartView"/>
        </item>
        <item row="0" column="2" colspan="2">
         <widget class="QLabel" name="inputlabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
